src/main/java/com/tabnine/binary/fetch/BinaryDownloader.java

```diff
@@ -22,13 +23,15 @@ public BinaryDownloader(TempBinaryValidator tempBinaryValidator) {
         this.tempBinaryValidator = tempBinaryValidator;
     }
 
-    public void downloadBinary(String version) throws FailedToDownloadException {
-        Path destination = versionFullPath(version);
+    public Optional<BinaryVersion> downloadBinary(String version) {
+        String destination = versionFullPath(version);
         Path tempDestination = Paths.get(format("%s.download.%s", destination, UUID.randomUUID()));
 
         try {
             if (!tempDestination.getParent().toFile().mkdirs()) {
-                throw new FailedToDownloadException("Could not create the required directories for " + tempDestination.toString());
+                Logger.getInstance(getClass()).warn(format("Could not create the required directories for %s", tempDestination));
```

Comment:
This could be an error (for the user to see), but I'm not sure the product would approve @dimacodota